lib/CL/devices/remote/communication.c

```diff
@@ -773,12 +773,46 @@ pocl_remote_reader_pthread (void *aa)
       // READ EXTRA DATA
       if (running_cmd->reply.data_size > 0)
         {
-          assert (running_cmd->rep_extra_data != NULL);
-          assert (running_cmd->rep_extra_size >= running_cmd->reply.data_size);
-          running_cmd->rep_extra_size = running_cmd->reply.data_size;
-          readb = read_full (fd, running_cmd->rep_extra_data,
-                             running_cmd->reply.data_size, remote);
-          CHECK_READ (readb);
+          if (rep.message_type == MessageType_DeviceInfoReply)
```

Comment:
I would just add a "rep_strings" pointer to running_cmd and make this conditional on rep.strings_size > 0 instead.